lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -163,16 +170,23 @@ putWallet
     -> ApiT WalletId
     -> WalletPutData
     -> Handler ApiWallet
-putWallet _ _ _ =
-    throwM err501
+putWallet w (ApiT wid) body = do
+    case body ^. #name of
+        Nothing ->
+            return ()
+        Just (ApiT wName) ->
+            liftHandler $ W.updateWallet w wid (\meta -> meta { name = wName })
+    getWallet w (ApiT wid)
```

Comment:
hm - could we have race condition between L178 and L179?

 * on line L178 wallet gets updated
 * another request is made which changes content of the same wid to <something_else>
 * on line L179 we return <something_else> which is not expected

Reply:
In more details:
 * thread A wants to update a wallet to "Foo" but is stoped just before L179
 * thread B runs and updates a wallet. It changes `name` to "Bar"
 * thread A is resumed and it runs L179. As a result it returns "Bar" (but what A expects is tu return "Foo")

Reply:
ps - its no big deal for wallet name - and in fact request for updating a wallet is not something that will likely be at a race condition situation a lot (as usually single user has access to single wallet)

Reply:
> hm - could we have race condition between L178 and L179?

Yes. Could be. Though it doesn't matter much. I mean, the response will be slightly off in the case of concurrent calls. I don't think it's really important for things like metadata as long as, the internal DB state is updated correctly (which is the case with the lock mechanism).